app/code/Magento/CatalogGraphQl/Model/Resolver/Product/ProductFieldsSelector.php

```diff
@@ -36,25 +45,15 @@ public function __construct(FieldTranslator $fieldTranslator)
      * @param string $productNodeName
      * @return string[]
      */
-    public function getProductFieldsFromInfo(ResolveInfo $info, string $productNodeName = 'product') : array
+    public function getProductFieldsFromInfo(ResolveInfo $info, string $productNodeName = 'product'): array
     {
         $fieldNames = [];
         foreach ($info->fieldNodes as $node) {
             if ($node->name->value !== $productNodeName) {
                 continue;
             }
-            foreach ($node->selectionSet->selections as $selectionNode) {
-                if ($selectionNode->kind === NodeKind::INLINE_FRAGMENT) {
```

Comment:
this code does exist in other places and I feel we have broken something else by removing it
there's named fragments and inline fragments. Your bug was abut named fragments and I think we broke inline fragments as in the if after the foreach